autoload/vundle.vim

```diff
@@ -61,24 +61,30 @@ endif
 " :Plugin command in the vimrc.  It is not possible to do this automatically
 " because when loading the vimrc file no plugins where loaded yet.
 func! vundle#rc(...) abort
-  let g:bundle_dir = len(a:000) > 0 ? expand(a:1, 1) : expand('$HOME/.vim/bundle', 1)
-  let g:updated_bundles = []
-  let g:vundle_log = []
-  let g:vundle_changelog = ['Updated Plugins:']
+  if a:0 > 0
+    let g:vundle#bundle_dir = expand(a:1, 1)
+  endif
   call vundle#config#init()
 endf
 
 " Alternative to vundle#rc, offers speed up by modifying rtp only when end()
 " called later.
 func! vundle#begin(...) abort
-  let g:vundle_lazy_load = 1
+  let g:vundle#lazy_load = 1
   call call('vundle#rc', a:000)
 endf
 
 " Finishes putting plugins on the rtp.
 func! vundle#end(...) abort
-  unlet g:vundle_lazy_load
+  unlet g:vundle#lazy_load
   call vundle#config#activate_bundles()
 endf
 
+" Initialize some global variables used by Vundle.
+let vundle#bundle_dir = expand('$HOME/.vim/bundle', 1)
```

Comment:
Curious here: Why bother giving this a sensible value when we **always** will initialise it in `#rc`?